test/Api.Test/Controllers/OrganizationDomainControllerTests.cs

```diff
@@ -0,0 +1,220 @@
+﻿using Bit.Api.Controllers;
+using Bit.Api.Models.Request;
+using Bit.Api.Models.Response;
+using Bit.Api.Models.Response.Organizations;
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.OrganizationFeatures.OrganizationDomains.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using Xunit;
+using Organization = Bit.Core.Entities.Organization;
+using OrganizationDomain = Bit.Core.Entities.OrganizationDomain;
+
+namespace Bit.Api.Test.Controllers;
+
+[ControllerCustomize(typeof(OrganizationDomainController))]
+[SutProviderCustomize]
+public class OrganizationDomainControllerTests
+{
+    [Theory, BitAutoData]
+    public async Task Get_ShouldThrowUnauthorized_WhenOrgIdCannotManageSso(Guid orgId,
+        SutProvider<OrganizationDomainController> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().ManageSso(orgId).Returns(false);
+
+        var requestAction = async () => await sutProvider.Sut.Get(orgId.ToString());
+
+        await Assert.ThrowsAsync<UnauthorizedAccessException>(requestAction);
+    }
+
+    [Theory, BitAutoData]
+    public async Task Get_ShouldNotFound_WhenOrganizationDoesNotExist(Guid orgId,
+        SutProvider<OrganizationDomainController> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().ManageSso(orgId).Returns(true);
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(orgId).ReturnsNull();
+
+        var requestAction = async () => await sutProvider.Sut.Get(orgId.ToString());
+
+        await Assert.ThrowsAsync<NotFoundException>(requestAction);
+    }
+
+    [Theory, BitAutoData]
+    public async Task Get_ShouldReturnOrganizationDomainList_WhenOrgIdIsValid(Guid orgId,
+        SutProvider<OrganizationDomainController> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().ManageSso(orgId).Returns(true);
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(orgId).Returns(new Organization());
+
+        var result = await sutProvider.Sut.Get(orgId.ToString());
+
+        Assert.IsType<ListResponseModel<OrganizationDomainResponseModel>>(result);
```

Comment:
Is it a slightly better test to initialize the org with an id / prop that you know and then assert that the result contains the object you created and expected to be returned?

Reply:
Nice suggestion